tools/parse-ramdisk-logs.py

```diff
@@ -94,29 +79,55 @@ def __call__(self, line):
             self.visitor(line)
 
 
-with open(args.source) as fp:
-    for line in fp:
-        line = line.strip()
+def parse(source: str, dest: str):
+    """Parse the log file at source and unpack all entries at dest."""
+    # Visitor is an object that is currently responsible for handling new
+    # lines.  These objects are hierarchical: top level is host (Ironic node),
+    # second level is specific source (file or command). Initially, visitor is
+    # None: such lines are skipped.
+    visitor = None
```

Comment:
```suggestion
    visitor: Optional[VisitorEntry] = None
```

and `from typing import Optional` on top.